osimpipeline/vital_tasks.py

```diff
@@ -542,7 +507,327 @@ def __init__(self, trial, ik_setup_task):
+                'Actuation_power.sto', 'Actuation_speed.sto',
+                'avgResiduals.txt', 'controls.sto', 'controls.xml',
+                'Kinematics_dudt.sto', 'Kinematics_q.sto', 'Kinematics_u.sto',
+                'pErr.sto', 'states.sto']:
+            self.targets += ['%s/results/%s_%s_%s_%s' % (
+                self.path, self.study.name, trial.id, tool_folder, rra_output)]
+
+class TaskRRAModel(TaskRRA):
+    REGISTRY = []
+    def __init__(self, trial, rramodel_setup_task, cycle=None, 
+                 reenable_probes=False):
+        super(TaskRRAModel, self).__init__('rramodel', trial, cycle=cycle)
+        self.doc = "Run OpenSim's Residual Reduction Algorithm tool to create an adjusted model."
+
+        # Set file dependencies
+        self.file_dep += [rramodel_setup_task.results_setup_fpath,
+                          rramodel_setup_task.results_tasks_fpath,
+                          rramodel_setup_task.results_extloads_fpath, 
+                          rramodel_setup_task.res_act_fpath,
+                          ]
+        # Set targets
+        self.targets += [rramodel_setup_task.adjusted_model_fpath]
+
+        # Deal with the fact that this operation would otherwise overwrite
+        # the valuable RRA log.
+        cur_log = '%s/out.log' % self.path
+        rra_log = '%s/out_rra.log' % self.path
+
+        def copylog():
+            if os.path.exists(cur_log): shutil.copyfile(cur_log, rra_log)
+        def removelog():
+            if os.path.exists(cur_log): os.remove(cur_log)
+        def renamelog():
+            if os.path.exists(rra_log): os.rename(rra_log, cur_log)
+        self.actions += [
+                copylog,
+                removelog,
+                renamelog,
+                ]
+
+        if reenable_probes:
+            self.actions += [self.reenable_probes]
+
+    def reenable_probes(self, rramodel_setup_task):
+        # RRA disables the probes; let's re-enable them.
+        import opensim
+        m = opensim.Model(rramodel_setup_task.adjusted_model_fpath)
+        from perimysium.modeling import enable_probes
+        enable_probes(rramodel_setup_task.adjusted_model_fpath)
+
+class TaskRRAKin(TaskRRA):
+    REGISTRY = []
+    def __init__(self, trial, rrakin_setup_task, cycle=None):
+        super(TaskRRAKin, self).__init__('rrakin', trial, cycle=cycle)
+        self.doc = "Run OpenSim's Residual Reduction Algorithm tool to adjust kinematics, using the model from TaskRRAModel."
 
+        # Set file dependencies
+        self.file_dep += [
+                rrakin_setup_task.adjusted_model_fpath,
+                rrakin_setup_task.results_setup_fpath,
+                rrakin_setup_task.results_tasks_fpath,
+                rrakin_setup_task.results_extloads_fpath,
+                ]
+
+class TaskCMCSetup(task.SetupTask):
+    REGISTRY = []
+    def __init__(self, trial, 
+                 adjusted_model=None,
+                 des_kinematics='rramodel', 
```

Comment:
Did you choose this default value because you plan on not using `rrakin`? In my workflow, `rramodel`'s resulting kinematics were not good to use, and I would have preferred `ik` or `rrakin` as the default value.